client/ui/actions/addon-groups.js

```diff
@@ -0,0 +1,171 @@
+import { browserHistory } from 'react-router';
+
+import apiFetch from '../utils/apiFetch';
+
+
+// AddonGroups list
+export const FETCH_ADDON_GROUPS_REQUEST = 'FETCH_ADDON_GROUPS_REQUEST';
+export const FETCH_ADDON_GROUPS_SUCCESS = 'FETCH_ADDON_GROUPS_SUCCESS';
+export const FETCH_ADDON_GROUPS_FAILURE = 'FETCH_ADDON_GROUPS_FAILURE';
+export const RESET_ADDON_GROUPS = 'RESET_ADDON_GROUPS';
+
+// Create an addon group
+export const CREATE_ADDON_GROUP_REQUEST = 'CREATE_ADDON_GROUP_REQUEST';
+export const CREATE_ADDON_GROUP_SUCCESS = 'CREATE_ADDON_GROUP_SUCCESS';
+export const CREATE_ADDON_GROUP_FAILURE = 'CREATE_ADDON_GROUP_FAILURE';
+export const RESET_NEW_ADDON_GROUP = 'RESET_NEW_ADDON_GROUP';
+
+// Fetch a single addon group
+export const FETCH_ADDON_GROUP_REQUEST = 'FETCH_ADDON_GROUP_REQUEST';
+export const FETCH_ADDON_GROUP_SUCCESS = 'FETCH_ADDON_GROUP_SUCCESS';
+export const FETCH_ADDON_GROUP_FAILURE = 'FETCH_ADDON_GROUP_FAILURE';
+export const RESET_ADDON_GROUP = 'RESET_ADDON_GROUP';
+
+// Update an addon group
+export const UPDATE_ADDON_GROUP_REQUEST = 'UPDATE_ADDON_GROUP_REQUEST';
+export const UPDATE_ADDON_GROUP_SUCCESS = 'UPDATE_ADDON_GROUP_SUCCESS';
+export const UPDATE_ADDON_GROUP_FAILURE = 'UPDATE_ADDON_GROUP_FAILURE';
+export const RESET_UPDATE_ADDON_GROUP = 'RESET_UPDATE_ADDON_GROUP';
+
+
+function apiError(type, error) {
+  return {
+    type,
+    error,
+  };
+}
+
+function requestAddonGroups() {
+  return {
+    type: FETCH_ADDON_GROUPS_REQUEST,
+  };
+}
+
+function receivedAddonGroups(addonGroups) {
+  return {
+    type: FETCH_ADDON_GROUPS_SUCCESS,
+    addonGroups,
+  };
+}
+
+function requestCreateAddonGroup() {
+  return {
+    type: CREATE_ADDON_GROUP_REQUEST,
+  };
+}
+
+function receivedCreateAddonGroup(addonGroup, saveAndContinue) {
+  if (saveAndContinue) {
+    browserHistory.push(`/addon_groups/${addonGroup.id}/`);
+  } else {
+    browserHistory.push('/addon_groups/');
+  }
+
+  return {
+    type: CREATE_ADDON_GROUP_SUCCESS,
+    addonGroup,
+  };
+}
+
+function requestAddonGroup(pk) {
+  return {
+    type: FETCH_ADDON_GROUP_REQUEST,
+    pk,
+  };
+}
+
+function receivedAddonGroup(addonGroup) {
+  return {
+    type: FETCH_ADDON_GROUP_SUCCESS,
+    addonGroup,
+  };
+}
+
+function requestUpdateAddonGroup() {
+  return {
+    type: UPDATE_ADDON_GROUP_REQUEST,
+  };
+}
+
+function receivedUpdateAddonGroup(addonGroup, saveAndContinue) {
+  if (!saveAndContinue) {
+    browserHistory.push('/addon_groups/');
+  }
```

Comment:
Action creators should be pure, that is, have no side effects. I think it would be better to use react-router-redux's action creators for changing browserHistory, [described here](https://github.com/reactjs/react-router-redux#what-if-i-want-to-issue-navigation-events-via-redux-actions). This lets the redux devtools work correctly. We've found those very helpful on Normandy.